File: 프로그래머스/0/181866. 문자열 잘라서 정렬하기/문자열 잘라서 정렬하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<string> solution(string myString) {
    vector<string> answer;
    string tmp = "";
    for(char t : myString) {
        if(t == 'x') {
            if(tmp != "") {
                answer.push_back(tmp);
                tmp = "";
            }
            continue;
        }
        tmp += t;
    }
    if(tmp != "") {
        answer.push_back(tmp);
    }
    
    sort(answer.begin(), answer.end());
    return answer;
}

File: 프로그래머스/0/181866. 문자열 잘라서 정렬하기/문자열 잘라서 정렬하기.java
import java.util.*;
class Solution {
    public String[] solution(String myString) {
        String[] answer = myString.split("x");
        ArrayList<String> list = new ArrayList<>();
        for(String s : answer) {
            if(!s.equals("")) {
                list.add(s);
            }
        }
        Collections.sort(list);
        
        return list.toArray(new String[0]);
    }
}